aten/src/ATen/native/native_functions.yaml

```diff
@@ -760,6 +760,16 @@
 - func: transpose_(Tensor self, int64_t dim0, int64_t dim1) -> Tensor
   variants: method
 
+- func: flip(Tensor self, IntList dims) -> Tensor
+  dispatch:
+    CPU: flip_cpu
+    CUDA: flip_cuda
+
+- func: flip_backward(Tensor grad, IntList dims) -> Tensor
```

Comment:
Does this mean I don't need the declaration in native_functions.yaml ? I do have an entry in derivatives.yaml: 

+- name: flip(Tensor self, IntList dims)
+  self: flip_backward(grad, dims)

Am I doing this right?

Reply:
`fn` should be put into `native_functions.yaml` when `fn` 
1. needs custom backward definition (either autograd doesn't know how to backward or it needs more efficient custom backward)
2. needs generated python bindings.

For a native function, there are usually three ways to define backward:
1. a simple formula in `derivatives.yaml`
```yaml
- name: atan(Tensor self)
  self: grad / (self * self + 1)
```
2. calling a complex function in `derivatives.yaml`
```yaml
- name: det(Tensor self)
  self: det_backward(grad, self, result)
```
with the function (`det_backward` here) defined usually in `Functions.cpp` (or as another native function if it needs custom backward)
3. without any entry. autograd will attempt to use the function calls in `fn` to do backward, e.g. `stft`. Note that this will be problematic if you do any low-level manipulations, because those are not tracked by autograd.